daemons/pacemakerd/pacemakerd.c

```diff
@@ -1170,7 +1236,13 @@ main(int argc, char **argv)
     mainloop_add_signal(SIGTERM, pcmk_shutdown);
     mainloop_add_signal(SIGINT, pcmk_shutdown);
 
-    init_children_processes();
+    if (running_with_sbd) {
+        pacemakerd_state = XML_PING_ATTR_PACEMAKERDSTATE_WAITPING;
+        startup_trigger = mainloop_add_trigger(G_PRIORITY_HIGH, init_children_processes, NULL);
```

Comment:
Don't we also need some sort of sbd version check? Otherwise pacemakerd will wait forever if an old sbd is running.

Reply:
Was thinking to do that via package dependencies / conflicts.
For upstream I was just thinking if I should bump sbd version so that we can put a conflicts in the upstream spec-file.

Reply:
I'd rather not make the new sbd version a hard requirement upstream, it would be better if we could look for a header file constant or something

Reply:
To be honest making it somehow run here is a bit dangerous to end up in a situation where the user thinks he has a combination using the handshaking because one of the sides does support it but in fact they somehow negotiated not to use it.
The combination of old sbd + new pacemaker actually doesn't worry me that much as it keeps the resources down and thus safe from split-brain.
The other way round sbd should be missing the symbols of the pacemakerd-api and thus not come up why systemd should keep pacemakerd down. So this way round it should be safe as well.

Reply:
> The combination of old sbd + new pacemaker actually doesn't worry me that much as it keeps the resources down and thus safe from split-brain.

That does bother me though. Someone could do what they think is a small pacemaker upgrade and suddenly their cluster won't run resources. That does make me think that in any case, we should log what pacemaker is waiting for. Perhaps we should make the waiting code conditional on a defined constant for now (2.0.5 release) and then make it standard for 2.1.0 (which should be the next release after that -- the minor-version bump should be a bigger signal to pay attention to the changes, and we can require new dependency versions).

> The other way round sbd should be missing the symbols of the pacemakerd-api and thus not come up why systemd should keep pacemakerd down. So this way round it should be safe as well.

If sbd is checking at configure-time for pacemaker support, that won't be a problem (unless someone upgrades pacemaker later, which would make pacemaker wait for a ping that never comes, but then that's like the previous situation and handled by whatever we decide there).

Reply:
Using a header this way round isn't easy as sbd is no build requirement for pacemaker.
The other way round I'm checking for the pacemakerd-api to be present with the pacemaker-devel-packages.

Reply:
Yep, I'm leaning to disabling the wait by default until 2.1.0 (luckily that's on the near horizon)

Reply:
Other idea coming to my mind:
What if we make it a config-option in sbd in /etc/sysconfig/sbd?
Pacemaker can react on that. If waiting isn't enabled pacemaker would
issue a big warning.
If sbd is using the pacemakerd-api but the waiting isn't enabled in /etc/sysconfig/sbd
we would log a big warning from sbd-side.

Reply:
That could work